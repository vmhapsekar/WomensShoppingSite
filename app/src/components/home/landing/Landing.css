@import "../../../styles/colors.css";
@import '../../../styles/mediaQueries.css';

.landing-section {
  position: relative;
  top: 0px;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
  background-size: cover;
}
