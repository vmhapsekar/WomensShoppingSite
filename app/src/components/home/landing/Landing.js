import React, {Component} from 'react'
import PropTypes from 'prop-types'
import {Grid, GRow, GCol} from '../../common/grid/Grid'
import Header from '../../common/header/Header'
import {SIZE_FILTER} from '../../../constants/appConstants'
import {filterProductsOnSize} from '../../../utils/productUtils'
import Products from '../products/Products'

class Landing extends Component {
  constructor (props) {
    super(props)
    this.state = {
      products: [],
      selectedSize: 0,
      filteredProducts: []
    }
    this.setFilter = this.setFilter.bind(this)
  }

  componentDidMount () {
    this.props.fetchAllProductData()
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.productData.productDataLoaded && nextProps.productData.fetchProductSuccess) {
      this.setState({products: nextProps.productData.products})
    }
  }

  setFilter (event) {
    let selectedSize = parseInt(event.target.value)
    if (selectedSize === 0) {
      this.setState({selectedSize: selectedSize})
    } else {
      this.setState({
        filteredProducts: filterProductsOnSize(this.props.productData.products, SIZE_FILTER[selectedSize].label),
        selectedSize: selectedSize})
    }
  }

  render () {
    let products = (this.state.selectedSize !== 0) ? this.state.filteredProducts : this.state.products
    let showProducts = this.props.productData.productDataLoaded && this.props.productData.fetchProductSuccess
    return (
      <Grid fluid>
        <GRow>
          <GCol xs={12} sm={12} md={12} lg={12}>
              <Header title="Women's tops" filter={SIZE_FILTER} setFilter={this.setFilter}/>
          </GCol>
          <GCol xs={12} sm={12} md={12} lg={12}>
            {showProducts ? <Products products={products}></Products> : null}
          </GCol>
        </GRow>
      </Grid>
    )
  }
}

Landing.propTypes = {
  fetchAllProductData: PropTypes.func.isRequired
}

export default Landing
