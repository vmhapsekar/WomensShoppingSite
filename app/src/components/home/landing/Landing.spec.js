import React from 'react'
import Landing from './Landing'
import {shallowToJson} from 'enzyme-to-json'
import Enzyme, {shallow} from 'enzyme'
const products = require('../../../assets/toolkit/data/products.json')

global.requestAnimationFrme = function (callback) {
  setTimeout(callback, 0)
}

describe('Landing', () => {

  const props = {
    fetchAllProductData: jest.fn(),
    productData: {productDataLoaded: false}
  }
  test(`should render <Header> and <Products>`, () => {
    const component = shallow(<Landing {...props} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })
  test(`if product data is loaded set products in state`, () => {
    const props1 = {
      fetchAllProductData: jest.fn(),
      productData: {productDataLoaded: true, fetchProductSuccess: true, products: ['product1', 'product2']}
    }
    const component = shallow(<Landing {...props} />)
    component.update()
    component.setProps(props1)
  })
  test(`if product data is loaded and if not do not render Products`, () => {
    const props1 = {
      fetchAllProductData: jest.fn(),
      productData: {productDataLoaded: false, fetchProductSuccess: false}
    }
    const component = shallow(<Landing {...props} />)
    component.update()
    component.setProps(props1)
  })
  test(`if products are filtered on change of size`, () => {
    const props1 = {
      fetchAllProductData: jest.fn(),
      productData: {productDataLoaded: true, fetchProductSuccess: true, products: products}
    }
    const setFilter = jest.spyOn(
      Landing.prototype,
      'setFilter'
    )
    const mockedEvent = {target: {value: "1"}}
    const component = shallow(<Landing {...props1} />)
    expect(component).toBeDefined()
    component.instance().setFilter(mockedEvent)
    expect(setFilter).toHaveBeenCalled()
  })
  test(`change the view when filter option set to 0`, () => {
    const props1 = {
      fetchAllProductData: jest.fn(),
      productData: {productDataLoaded: true, fetchProductSuccess: true, products: products}
    }
    const setFilter = jest.spyOn(
      Landing.prototype,
      'setFilter'
    )
    const mockedEvent = {target: {value: "0"}}
    const component = shallow(<Landing {...props1} />)
    expect(component).toBeDefined()
    component.instance().setFilter(mockedEvent)
    expect(setFilter).toHaveBeenCalled()
  })
})