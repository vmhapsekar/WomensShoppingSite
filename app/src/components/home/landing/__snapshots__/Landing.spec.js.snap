// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render <Header> and <Products> 1`] = `
<Grid
  fluid={true}
>
  <GRow>
    <GCol
      lg={12}
      md={12}
      sm={12}
      xs={12}
    >
      <Header
        filter={
          Array [
            Object {
              "id": 0,
              "label": "Filter by size",
            },
            Object {
              "id": 1,
              "label": "XS",
            },
            Object {
              "id": 2,
              "label": "S",
            },
            Object {
              "id": 3,
              "label": "M",
            },
            Object {
              "id": 4,
              "label": "L",
            },
            Object {
              "id": 5,
              "label": "XL",
            },
          ]
        }
        setFilter={[Function]}
        title="Women's tops"
      />
    </GCol>
    <GCol
      lg={12}
      md={12}
      sm={12}
      xs={12}
    />
  </GRow>
</Grid>
`;
